### ISSUE TITLE
Casualties Cap: Multiple endings possible

### ISSUE THREAD

**Issue by 8h2a** [ferstaberinde/F3#776]
In our adversarials we often use something like this:

```
[BLUFOR,100,1] execVM "f\casualtiesCap\f_CasualtiesCapCheck.sqf";
[OPFOR,100,2] execVM "f\casualtiesCap\f_CasualtiesCapCheck.sqf";
```

Now if BLUFFOR dies, it triggers ending 1, but if - at a later time (e.g. a few seconds later) - OPFOR dies, it triggers the other ending. So the player actually sees two contradicting endings. That's probably not what most missionmakers/players want.

Maybe it's preferred to only have one ending and not both.
Do you think this should be a feature for f3 casualtiesCap?

**Comment by Wolfenswan** [comment]
I'm not sure I follow. If a numbered ending fires the mission is simply over, there's no way to continue playing.

If you want custom things to happen when a side is wiped out you can already do so using code instead of a number:

```
[BLUFOR,100,{hint 'Blufor dead!'}] execVM "f\casualtiesCap\f_CasualtiesCapCheck.sqf";
[OPFOR,100,{hint 'Opfor dead!'}] execVM "f\casualtiesCap\f_CasualtiesCapCheck.sqf";
```

**Comment by Wolfenswan** [comment]
Oh, now I understand. You mean if OPFOR dies within a few seconds after BLUFOR, the OPFOR ending "overrides" the BLUFOR ending?

I'll look into it, it's possible it's sort of an engine limitation but maybe using a variable check that kind of thing can be prevented.

**Comment by 8h2a** [comment]
Yes, exactly. I was thinking about a variable that checks if the Casualty Cap already triggered an ending or not, and then prevents other endings.
But this should probably only happen when the third argument is a number. Or it should be configureable? I'm not sure if people use custom code and rely on the fact that Casualty Cap can trigger multiple times.
e.g. one could do this:

```
[BLUFOR,80,{hint 'Blufor almost dead!'}] execVM "f\casualtiesCap\f_CasualtiesCapCheck.sqf";
[BLUFOR,100,1] execVM "f\casualtiesCap\f_CasualtiesCapCheck.sqf";
```

**Comment by Wolfenswan** [comment]
One thing you could already do is check inside mpEndReceiver.sqf:

```
switch (_ending) do
{
// Ending #1 (OPFOR dead)
    case 1:
    {
// Place any custom code for ending #1 after this line:
if ( {side _x == west && alive _x} count playableUnits == 0 ) exitWith {};


// Do not allow custom code for ending #1 to continue after this comment.
    };
// Ending #2 (BLUFOR dead)
    case 2:
    {
// Place any custom code for ending #2 after this line:
if ( {side _x == east && alive _x} count playableUnits == 0 ) exitWith {};


// Do not allow custom code for ending #2 to continue after this comment.
    };
```

Not 100% sure if exitWith considers switch do a scope or if it exists the entire script.

**Comment by LISTINGS09** [comment]
We got around this issue by putting a variable into fn_mpEnd.sqf, which ensures only the first ending call the server receives will be sent out.

`if (isNil "f_var_EndingTriggered") then {
    [_this,"f_fnc_mpEndReceiver",true] spawn BIS_fnc_MP;
};

f_var_EndingTriggered = true;`
